storage/innobase/log/log0recv.cc

```diff
@@ -1467,13 +1474,15 @@ recv_find_max_checkpoint(ulint* max_field)
 	ib_uint64_t	checkpoint_no;
 	ulint		field;
 	byte*		buf;
+	dberr_t		err;
```

Comment:
Instead of declaring a local variable for the entire function, I’d narrow the scope of the variable to each `if` block that is invoking `log_sys.log.read()`.